llvm/lib/IR/Attributes.cpp

```diff
@@ -286,6 +286,12 @@ Attribute Attribute::getWithNoFPClass(LLVMContext &Context,
   return get(Context, NoFPClass, ClassMask);
 }
 
+Attribute Attribute::getWithRange(LLVMContext &Context,
+                                  const ConstantRange &CR) {
+  assert(!CR.isFullSet() && "Range must not be full!");
```

Comment:
I think it would be okay to just add this directly into the generic ctor, rather than specific to `range`. I would expect that other range-based attributes added in the future will not allow full ranges either. (If they do, we can separate it out at the time.)

Reply:
yes that seems resonable.